src/plot_api/plot_schema.js

```diff
@@ -460,11 +460,22 @@ function getTraceAttributes(type) {
     // make 'type' the first attribute in the object
     attributes.type = null;
 
+
+    var copyBaseAttributes = extendDeepAll({}, baseAttributes),
+        copyModuleAttributes = extendDeepAll({}, _module.attributes);
+
+    // prune global-level trace attributes that are already defined in a trace
+    exports.crawl(copyModuleAttributes, function(attr, attrName, attrs, level, fullAttrString) {
+        delete copyBaseAttributes[fullAttrString];
```

Comment:
`fullAttrString` might look like `marker.line.width` in general, so we'll need to use [`Lib.nestedProperty`](https://github.com/plotly/plotly.js/blob/master/src/lib/nested_property.js) here.

Reply:
Thank you @etpinard! Commit 7999517 fixes this :)